backend/src/main/java/server/handlers/QueueAddHandler.java

```diff
@@ -0,0 +1,67 @@
+package server.handlers;
+
+import Matchmaking.Player;
+import Matchmaking.Position;
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.Moshi;
+import com.squareup.moshi.Types;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import server.Server;
+import server.exceptions.ItemAlreadyExistsException;
+import server.exceptions.NoItemFoundException;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+
+public class QueueAddHandler implements Route {
+
+  private Server server;
+
+  public QueueAddHandler(Server server) {
+    this.server = server;
+  }
+
+  /**
+   * Invoked when a request is made on this route's corresponding path e.g. '/hello'
+   *
+   * @param request  The request object providing information about the HTTP request
+   * @param response The response object providing functionality for modifying the response
+   * @return The content to be set in the response
+   * @throws Exception implementation can choose to throw exception
+   */
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    Moshi moshi = new Moshi.Builder().build();
+    Type mapStringObject = Types.newParameterizedType(Map.class, String.class, Object.class);
+    JsonAdapter<Map<String, Object>> adapter = moshi.adapter(mapStringObject);
+    HashMap<String, Object> responseMap = new HashMap<>();
+
+    String playerID;
+    try {
```

Comment:
I just found out that this way of handling queries might now work, so I may change this a bit. Overall though it looks great!